install-libs.cmd

```diff
@@ -0,0 +1,11 @@
+rem This assumes you have nuget.exe installed and are running from the same directory
+rem It used to be possible to install these via packages.config but it doesn't seem to work in VS2017 after moving to ProjectReferences
+md packages
+cd packages
+nuget install rapidjson.v110
+nuget install tinyformat
```

Comment:
I really don't know how git submodules work. In the past I would use NuGet Packages to consume third party libs. The rapidjson retrieved is the newest one, not the beta version.

tinyformat is type-safe replacement of printf, it uses output streams. You could use for ofstream, ostringstream, ostream (cout,cerr,etc), or if leave the stream part out (like a normal printf) it'll either just print it out (tfm::printf) or return a string (tfm::format). It helped a lot when I was trying make a type safe version of that RA_LOG. It eliminates the need of using statically allocated buffers. 

If tl;dr
It's basically printf syntax (specific flags don't matter) with modern C++.

Reply:
The current RapidJSON used does not have the capacity for what I'm trying to do.

They aren't target specific. Well if there was a target it would be native,Version=v0.0 (according to NuGet).

Reply:
You can use the git command `git clone --recurse-submodules` to clone a repository and it's included submodules.

I don't see the purpose of the inclusion of `tinyformat`. Personally I rather exclude it altogether, but if you absolutely need to have it, then I would go the submodule route along with RapidJSON and ditch the script.